### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by swissspidy**
* More modern syntax
* Use single namespace file instead of god class
* Use GitHub Actions instead of travis
* Introducing linting

### CODE REVIEW

**Comment by grappler** [comment]
Test Email mit Kommentar und Passwort zurücksetzten.
image

**Comment by ocean90** [comment]
Can you set up the [linting workflow](https://github.com/wearerequired/.github/blob/master/workflow-templates/linting.yml) (~without eslint/stylelint~) please?

**Comment by grappler** [comment]
> Can you set up the linting workflow (without eslint/stylelint) please?

You mean deactivate eslint/stylelint or create a separate workflow for them?

**Comment by ocean90** [comment]
Ah, I thought the plugin has no CSS/JS files. In this case including the workflow as is would be nice.

**Comment by grappler** [comment]
I have added it but it has not triggered. There are a few other changes that I would like to make like remove grunt but I plan to do that in a separate PR.

**Comment by ocean90** [comment]
I don't know why it doesn't run. 😞

**Comment by grappler** [comment]
It might only run when creating a PR. The other task runs on push.

**Comment by required-com** [comment]
`pull_request` should also include the `synchronized` event which is for every new commit on a PR.

**Comment by ocean90** [comment]
`pull_request` doesn't run if the PR has merge conflicts…

**Comment by grappler** [comment]
@ocean90 I have fixed most of the issues. There are only "2 errors and 8 warnings" left. I am not sure what to do with them. Would love to have your feedback.

File: inc/Message/Section.php [hunk 1]
```diff
@@ -66,12 +59,11 @@ abstract class Section implements MessageInterface {
 	 * Sets the current user.
 	 *
 	 * @since  2.0.0
-	 * @access public
 	 *
-	 * @param array   $entries The message entries.
-	 * @param WP_User $user    The current user.
+	 * @param array    $entries The message entries.
+	 * @param \WP_User $user The current user.
```
Comment (ocean90) [inline inc/Message/Section.php]:
```suggestion
	 * @param \WP_User|null $user The current user.
```

File: inc/Message/CoreUpdate.php [hunk 2]
```diff
@@ -29,13 +26,12 @@ class CoreUpdate extends Section {
 	 * Constructor.
 	 *
 	 * @since  2.0.0
-	 * @access protected
 	 *
-	 * @param array   $entries The core update entries.
-	 * @param WP_User $user    The current user.
-	 * @param string  $event   The current event.
+	 * @param array    $entries The core update entries.
+	 * @param \WP_User $user The current user.
```
Comment (ocean90) [inline inc/Message/CoreUpdate.php]:
```suggestion
	 * @param \WP_User|null $user The current user.
```

**Review by ocean90 — CHANGES_REQUESTED** [review]
The two errors are due to the incorrect `@param` tag.

For the `current_time()` warnings we'd have to check each usage whether the current time in UTC or in the current timezone needs to be used. If UTC is fine we can switch to `time()`. The others can be ignored with the `phpcs:ignore` comment which should also explain why it's ignored.

**Comment by grappler** [comment]
Thanks @ocean90!

I have changed all of the times to UTC. There maybe an issue for the first emails after upgrading. I am not sure how best to mitigate this.

File: inc/Message/CoreUpdate.php [hunk 3]
```diff
@@ -29,13 +26,12 @@ class CoreUpdate extends Section {
...
 	 *
 	 * @since  2.0.0
-	 * @access protected
 	 *
-	 * @param array   $entries The core update entries.
-	 * @param WP_User $user    The current user.
-	 * @param string  $event   The current event.
+	 * @param array         $entries The core update entries.
+	 * @param \WP_User|null $user The current user.
+	 * @param string        $event   The current event.
```
Comment (ocean90) [inline inc/Message/CoreUpdate.php:30]:
```suggestion
	 * @param array         $entries The core update entries.
	 * @param \WP_User|null $user    The current user.
	 * @param string        $event   The current event.
```

File: template-tags/pluggable.php [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+		// Send instant email only to User (Modification 2).
+		$notify = 'user';
+
+		// The blogname option is escaped with esc_html() on the way into the database in sanitize_option().
+		// We want to reverse this for the plain text arena of emails.
+		$blogname = wp_specialchars_decode( get_option( 'blogname' ), ENT_QUOTES );
+
+		if ( 'user' !== $notify ) {
+			$switched_locale = switch_to_locale( get_locale() );
```
Comment (ocean90) [inline template-tags/pluggable.php:57]: This function was introduced in WordPress 4.7.0, but the current plugin requirement is 4.4.0. Let's update the "Requires at least" parameter.

File: digest.php [hunk 5]
```diff
@@ -12,8 +10,6 @@
  * License:     GPLv2+
  * Text Domain: digest
  * Domain Path: /languages
```
Comment (ocean90) [inline digest.php]: The updated PHP and WP requirements need to be added to the [plugin headers](https://developer.wordpress.org/plugins/plugin-basics/header-requirements/#header-fields).

File: inc/namespace.php [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+	register_activation_hook( __FILE__, __NAMESPACE__ . '\\activate_plugin' );
+	register_deactivation_hook( __FILE__, __NAMESPACE__ . '\\deactivate_plugin' );
+}
+
+/**
+ * Load plugin textdomain.
+ */
+function load_textdomain() {
+	load_plugin_textdomain( 'digest' );
+}
```
Comment (ocean90) [inline inc/namespace.php]:
```suggestion
```

This can be removed once the WP requirement has been raised to 4.7.

**Review by ocean90 — CHANGES_REQUESTED** [review]

**Review by ocean90 — APPROVED** [review]

File: phpcs.xml.dist [hunk 7]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
+<ruleset name="digest">
+	<description>Coding Standard for the "Digest" plugin.</description>
+
+	<file>.</file>
+
+	<rule ref="Required">
+		<!-- Not compatable with PHP 5.6 -->
```
Comment (ocean90) [inline phpcs.xml.dist]:
```suggestion
		<!-- Not compatible with PHP 5.6 -->
```

**Review by ocean90 — APPROVED** [review]

**Pull request merged**